lib/fastlane/plugin/ddg_apple_automation/actions/asana_get_release_automation_subtask_id.rb

```diff
@@ -0,0 +1,72 @@
+require "fastlane/action"
+require "fastlane_core/configuration/config_item"
+require "httparty"
+require "json"
+require "time"
+require_relative "../helper/ddg_apple_automation_helper"
+require_relative "../helper/github_actions_helper"
+require_relative "asana_extract_task_id_action"
+require_relative "asana_extract_task_assignee_action"
+
+module Fastlane
+  module Actions
+    class AsanaGetReleaseAutomationSubtaskIdAction < Action
+      def self.run(params)
+        task_url = params[:task_url]
+        token = params[:asana_access_token]
+
+        task_id = AsanaExtractTaskIdAction.run(task_url: task_url, asana_access_token: token)
+
+        # Fetch release task assignee and set GHA output
+        # TODO: To be reworked for local execution
+        AsanaExtractTaskAssigneeAction.run(task_id: task_id, asana_access_token: token)
+
+        url = Helper::DdgAppleAutomationHelper::ASANA_API_URL + "/tasks/#{task_id}/subtasks?opt_fields=name,created_at"
+        response = HTTParty.get(url, headers: { 'Authorization' => "Bearer #{token}" })
+
+        if response.success?
+          data = response.parsed_response['data']
+          automation_subtask_id = data
+                                  .find_all { |hash| hash['name'] == 'Automation' }
+                                  &.min_by { |x| Time.parse(x['created_at']) } # Get the oldest 'Automation' subtask
+                                  &.dig('gid')
```

Comment:
perhaps it makes sense to put into a separate private method, like so:

def self.find_oldest_automation_subtask(data)
  data
    .select { |hash| hash['name'] == 'Automation' }
    &.min_by { |x| Time.parse(x['created_at']) }
    &.dig('gid')
end

Reply:
Done!